lib/src/rules/parameter_assignments.dart

```diff
@@ -88,9 +136,33 @@ class _Visitor extends SimpleAstVisitor {
       // Getters don't have parameters.
       parameterList.parameters.forEach((e) {
         if (node.body.isPotentiallyMutatedInScope(e.element)) {
-          rule.reportLint(e);
+          _reportIfSimpleParameterOrWithDefaultValue(e, node);
         }
       });
     }
   }
+
+  void _reportIfSimpleParameterOrWithDefaultValue(
```

Comment:
I tested adding print statements to verify what was being invoked and I found that (with new changes):

* `_isDefaultFormalParameterWithoutDefaultValueMutated` in invoked only if `parameter is SimpleFormalParameter ||
        _isDefaultFormalParameterWithDefaultValue(parameter)` is `false`.
* `nonNullCoalescingAssignments` is traversed only if `assignmentsNodes.length <= 1` and since I am invoking toList now, `_isDefaultFormalParameterWithoutDefaultValueMutated` is not invoked again.

And IMO is easier to read :)